src/Doctrine/Types/CollectionValueObjectType.php

```diff
@@ -6,13 +6,22 @@
 
 use Doctrine\DBAL\Platforms\AbstractPlatform;
 use Doctrine\DBAL\Types\JsonType;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\DBAL\Types\Types;
 use Webmozart\Assert\Assert;
 use Yokai\DoctrineValueObject\CollectionValueObject;
 
-final class CollectionValueObjectType extends JsonType
+final class CollectionValueObjectType extends Type
```

Comment:
I thought it was a prerequisite of newer versions of Doctrine (class might have been final after all)
But after giving it a try, I'm not sure why you introduced such change
Can you explain me why?

Reply:
Found the answer myself, typehints were introduced in doctrine types at some point, so we cannot inherit it anymore, otherwise we get errors like
```
TypeError: Yokai\DoctrineValueObject\Doctrine\Types\StringValueObjectType::getInheritedType(): Return value must be of type Doctrine\DBAL\Types\StringType, Doctrine\DBAL\Types\JsonType returned
```

This is because StringValueObject and string are not covariant